### PR TITLE
Language ID Model

### PR DESCRIPTION
**Pull request by zrthxn**

### CODE REVIEW

File: src/seamless_communication/cli/m4t/classification_head/train.py [hunk 1]
```diff
@@ -184,28 +184,44 @@ def train(head: torch.nn.Module,
...
+                assert batch.speech_to_text.src_tokens is not None
+            logger.info(f"Epoch {epoch} Loss: {epoch_loss / len(dataloader)}")
+
+    except KeyboardInterrupt:
+        logger.info("Interrupted by user. Saving model state...")
+        torch.save({
+            'model_state_dict': head.state_dict(),
+            'optimizer_state_dict': optimizer.state_dict(),
+            'loss': loss,
+        }, "interrupted_model.pth")
```
Comment (zrthxn) [inline src/seamless_communication/cli/m4t/classification_head/train.py:220]: It would be better to save just the head state dict in one file so that if you were to use that to test something you could just load that whole file and not have to figure out that it also contains the optimizer and loss. Also since we don't have a way to resume training it isn't much use to save the optimizer state.

File: src/seamless_communication/cli/m4t/classification_head/train.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        logger.info("Model state saved. Exiting...")
+        exit()
+
+    logger.info(f"Final Loss: {sum(losslog) / len(losslog)}")
```
Comment (zrthxn) [inline src/seamless_communication/cli/m4t/classification_head/train.py:224]: This would give the average loss, which doesn't really make that much sense anyway
